### PR TITLE
Kaydunov shop list part2

### PR DESCRIPTION
**Pull request by KaydunovDenis**
## New fields and validation improvements
Task 1 for ShoppingListApplication
Branch: feature/1
1. Необходимо добавить в класс Product поля:   категория скидка описание
2. Необходимо дополнить проверку продукта:
- [x] цена продукта должна быть больше 0
- [x] скидка не может быть больше 100 процентов
- [x] название не может быть меньше 3 символов и больше 32

### Single Responsibility Principle
Task 2 for ShoppingListApplication
Необходимо сделать рефакторинг приложения на основе презентации Single Responsibility Principle.

### CODE REVIEW

File: homework/kaydunov_denis/shopping_list_application_part2/src/test/java/github/kaydunovdenis/bean/ProductTest.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+import org.slf4j.LoggerFactory;
+
+import java.math.BigDecimal;
+
+public class ProductTest {
+    private Product product;
+
+    @Before
+    public void setUp() {
+        product = getDefaultProduct();
```
Comment (max-java) [inline homework/kaydunov_denis/shopping_list_application_part2/src/test/java/github/kaydunovdenis/bean/ProductTest.java:17]: as you have fabric, no need to instantiate product every time here, no need to have product as a field in the test.

File: homework/kaydunov_denis/shopping_list_application_part2/src/test/java/github/kaydunovdenis/bean/ProductTest.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        String expect = "Notebook Acer 9302V-1";
+        String result = product.getName();
+        Assert.assertEquals(expect, result);
+    }
+
+    @Test
+    public void testGetId() {
+        Long expect = 1L;
+        Long result = product.getId();
+        Assert.assertEquals(expect, result);
```
Comment (max-java) [inline homework/kaydunov_denis/shopping_list_application_part2/src/test/java/github/kaydunovdenis/bean/ProductTest.java:31]: Here and next could be simplified by 
```
* {
    assertEquals(1L, getDefaultProduct().getId());
}

File: homework/kaydunov_denis/shopping_list_application_part2/src/test/java/github/kaydunovdenis/service/DiscountValidatorTest.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+import github.kaydunovdenis.bean.Product;
+import github.kaydunovdenis.bean.ProductTest;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.math.BigDecimal;
+
+import static org.junit.Assert.*;
```
Comment (max-java) [inline homework/kaydunov_denis/shopping_list_application_part2/src/test/java/github/kaydunovdenis/service/DiscountValidatorTest.java:11]: Use sonar!
```
static org.junit.Assert.*;
```
don't use static imports with *

File: homework/kaydunov_denis/shopping_list_application_part2/src/test/java/github/kaydunovdenis/service/DiscountValidatorTest.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        product.setDiscount(discount);
+        Assert.assertTrue(discountValidator.validate(product));
+
+        discount = new BigDecimal("100.0");
+        product.setDiscount(discount);
+        Assert.assertTrue(discountValidator.validate(product));
+
+        discount = new BigDecimal("111.32");
+        product.setDiscount(discount);
+        Assert.assertFalse(discountValidator.validate(product));
```
Comment (max-java) [inline homework/kaydunov_denis/shopping_list_application_part2/src/test/java/github/kaydunovdenis/service/DiscountValidatorTest.java:43]: should be divided into two cases: positive and negative scenario

remind me in lecture - I'll tell you more about it

File: homework/kaydunov_denis/shopping_list_application_part2/src/test/java/github/kaydunovdenis/service/PriceValidatorTest.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+import github.kaydunovdenis.bean.Product;
+import github.kaydunovdenis.bean.ProductTest;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.math.BigDecimal;
+
+import static org.junit.Assert.*;
```
Comment (max-java) [inline homework/kaydunov_denis/shopping_list_application_part2/src/test/java/github/kaydunovdenis/service/PriceValidatorTest.java:11]: static *

File: homework/kaydunov_denis/shopping_list_application_part2/src/test/java/github/kaydunovdenis/service/PriceValidatorTest.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+public class PriceValidatorTest {
+    private PriceValidator priceValidator;
+    private Product product;
+
+    @Before
+    public void setUp() {
+        priceValidator = new PriceValidator();
+        product = ProductTest.getDefaultProduct();
```
Comment (max-java) [inline homework/kaydunov_denis/shopping_list_application_part2/src/test/java/github/kaydunovdenis/service/PriceValidatorTest.java:20]: same for product as above

**Review by max-java — CHANGES_REQUESTED** [review]
ptal

File: homework/kaydunov_denis/shopping_list_application_part2/src/test/java/github/kaydunovdenis/bean/ProductTest.java [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+public class ProductTest {
+    private Product product;
+
+    @Before
+    public void setUp() {
+        product = getDefaultProduct();
+    }
+
+    @Test
+    public void testTestGetName() {
```
[the remark this answers is not in the document]
Comment (KaydunovDenis) [reply homework/kaydunov_denis/shopping_list_application_part2/src/test/java/github/kaydunovdenis/bean/ProductTest.java:21]: Junit genereted it. It seemed strange to me. I thought it should be so. Okay, I will fix it.

File: homework/kaydunov_denis/shopping_list_application_part2/src/main/java/github/kaydunovdenis/Constant.java [hunk 8]
```diff
@@ -0,0 +1,10 @@
+package github.kaydunovdenis;
+
+public class Constant {
```
[the remark this answers is not in the document]
Comment (KaydunovDenis) [reply homework/kaydunov_denis/shopping_list_application_part2/src/main/java/github/kaydunovdenis/Constant.java:3]: has been fixed

File: homework/kaydunov_denis/shopping_list_application_part2/src/main/java/github/kaydunovdenis/bean/Category.java [hunk 9]
```diff
@@ -0,0 +1,8 @@
+package github.kaydunovdenis.bean;
+
+public enum Category {
```
[the remark this answers is not in the document]
Comment (KaydunovDenis) [reply homework/kaydunov_denis/shopping_list_application_part2/src/main/java/github/kaydunovdenis/bean/Category.java:3]: has been fixed

File: homework/kaydunov_denis/shopping_list_application_part2/src/test/java/github/kaydunovdenis/bean/ProductTest.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (KaydunovDenis) [reply homework/kaydunov_denis/shopping_list_application_part2/src/test/java/github/kaydunovdenis/bean/ProductTest.java:21]: has been fixed

File: homework/kaydunov_denis/shopping_list_application_part2/src/test/java/github/kaydunovdenis/bean/ProductTest.java (hunk 1 above)
Comment (KaydunovDenis) [reply homework/kaydunov_denis/shopping_list_application_part2/src/test/java/github/kaydunovdenis/bean/ProductTest.java:17]: has been fixed

File: homework/kaydunov_denis/shopping_list_application_part2/src/test/java/github/kaydunovdenis/bean/ProductTest.java (hunk 2 above)
Comment (KaydunovDenis) [reply homework/kaydunov_denis/shopping_list_application_part2/src/test/java/github/kaydunovdenis/bean/ProductTest.java:31]: has been fixed

File: homework/kaydunov_denis/shopping_list_application_part2/src/test/java/github/kaydunovdenis/service/DiscountValidatorTest.java (hunk 3 above)
Comment (KaydunovDenis) [reply homework/kaydunov_denis/shopping_list_application_part2/src/test/java/github/kaydunovdenis/service/DiscountValidatorTest.java:11]: has been fixed

File: homework/kaydunov_denis/shopping_list_application_part2/src/test/java/github/kaydunovdenis/service/DiscountValidatorTest.java (hunk 4 above)
Comment (KaydunovDenis) [reply homework/kaydunov_denis/shopping_list_application_part2/src/test/java/github/kaydunovdenis/service/DiscountValidatorTest.java:43]: has been fixed

File: homework/kaydunov_denis/shopping_list_application_part2/src/test/java/github/kaydunovdenis/service/PriceValidatorTest.java (hunk 5 above)
Comment (KaydunovDenis) [reply homework/kaydunov_denis/shopping_list_application_part2/src/test/java/github/kaydunovdenis/service/PriceValidatorTest.java:11]: has been fixed

File: homework/kaydunov_denis/shopping_list_application_part2/src/test/java/github/kaydunovdenis/service/PriceValidatorTest.java (hunk 6 above)
Comment (KaydunovDenis) [reply homework/kaydunov_denis/shopping_list_application_part2/src/test/java/github/kaydunovdenis/service/PriceValidatorTest.java:20]: has been fixed

**Review by PavelBoPavel — APPROVED** [review]

**Review by trump-ny — APPROVED** [review]

**Review by trump-ny — APPROVED** [review]

**Pull request merged**